sidechain/storage/src/storage.rs

```diff
@@ -97,18 +101,74 @@ impl<SignedBlock: SignedBlockT> SidechainStorage<SignedBlock> {
 		self.db.get(block_hash)
 	}
 
-	/// update sidechain storage
+	/// Get all blocks following (i.e. children of) a specified blocks.
+	pub fn get_blocks_following(
+		&self,
+		block_hash: &BlockHash,
+		shard_identifier: &ShardIdentifierFor<SignedBlock>,
+	) -> Result<Vec<SignedBlock>> {
+		// Ensure we find the block in storage (otherwise we would return all blocks for a specific shard).
+		if self.get_block(block_hash)?.is_none() {
+			warn!("Could not find starting block in storage, returning empty vector");
+			return Ok(Vec::new())
+		}
+
+		// We get the latest block and then traverse the parents until we find our starting block.
+		let latest_block = match self.last_block_of_shard(shard_identifier) {
+			None =>
+				return Err(Error::LastBlockNotFound(
+					"Failed to find last block information".to_string(),
+				)),
+			Some(last_block_ref) => match self.get_block(&last_block_ref.hash)? {
+				None =>
+					return Err(Error::LastBlockNotFound(
+						"Failed to retrieve last block from storage".to_string(),
+					)),
+				Some(last_block) => last_block,
+			},
+		};
+
+		let mut current_block = latest_block;
+		let mut blocks_to_return = Vec::<SignedBlock>::new();
+		while &current_block.hash() != block_hash {
+			let parent_block_hash = current_block.block().parent_hash();
+
+			// Break in case we're at the genesis block.
+			if parent_block_hash == BlockHash::default() {
+				break
+			}
```

Comment:
Good question. I think in an ideal case this will never be necessary. The only time this will be of importance is in case the block we're looking for is not actually part of our stored blockchain. For example when the db was not cleaned properly before restarting.

Should I remove it?

Reply:
> The only time this will be of importance is in case the block we're looking for is not actually part of our stored blockchain

Which is what we check at line 111, no? If yes, I think you can remove it.